nbgrader/server_extensions/course_list/handlers.py

```diff
@@ -75,9 +75,13 @@ def check_for_local_formgrader(self, config):
             self.log.error(traceback.format_exc())
             raise gen.Return([])
 
+        coursedir = CourseDirectory(config=config)
+        if not coursedir.course_id:
```

Comment:
I am not sure this if statement has an effect, since the `gen.Return` is not actually raised. I think you can just remove it, since the javascript handles the case where the course id is missing.

Reply:
ah yes, good point! I've removed it, thanks

Reply:
Thanks!